controllers/admin/schema.js

```diff
@@ -147,7 +147,7 @@ router.delete('/remove_model', function(req, res, next) {
 	var appId = req._telepat.applicationId;
 	var modelName = req.body.model_name;
 
-	if (!tlib.apps[appId].schema[modelName]) {
+	if (!tlib.apps[appId].modelSchema[modelName]) {
```

Comment:
@adelinaenache modelSchema is only an object which contains functions (there's no schema there: https://github.com/telepat-io/telepat-models/blob/improvement/refactor-models/lib/Application.js#L240 ). Should use the **isValidModel()** function from the **modelSchema** object.